packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -477,6 +485,16 @@ export class NetworkMonitor {
     }
   }
 
+  // Wrapper function over this.subgraphDeployment that will throw an
+  // error on missing subgraph deployments
+  async requireSubgraphDeployment(ipfsHash: string): Promise<SubgraphDeployment> {
+    const subgraphDeployment = await this.subgraphDeployment(ipfsHash)
+    if (!subgraphDeployment) {
+      throw new Error(`Failed to locate subgraph deployment for ID ${ipfsHash}`)
+    }
+    return subgraphDeployment
+  }
+
```

Comment:
I think it's more readable to encapsulate error handling in this wrapper function.

I'm unsure if this pattern conforms to project error-handling principles, so I'd like to know if this is OK.

Reply:
I like that idea.
But I'm not sure IE020 would best represent this case (of not finding a deployment in the Network Subgraph.) 

It's a mix of IE020 and IE040, but still not entirely, as not finding the deployment details doesn't equate to a sync problem.


Should we have a dedicated error code for such a case?

(I'll prepare the PR for IE020, so we can discuss this detail later on)

Reply:
Handled in 8be6d17